File: JavaCourse/lab3/src/ua/lviv/iot/CompanyManager.java
package ua.lviv.iot;

import java.util.Comparator;
import java.util.LinkedList;
import java.util.List;

public class CompanyManager {
    private int totalCapacity = 0;
    private double totalLoadCapacity = 0;

    private List<Plane> planeList;

    public CompanyManager(List<Plane> planeList) {
        setPlaneList(planeList);
    }

    public List<Plane> searchFuelConsumption(double fuelConsumption) {
        List<Plane> result = new LinkedList<>();
        for (Plane aPlaneList : planeList) {
            if (aPlaneList.getFuelConsumption() == fuelConsumption) {
                result.add(aPlaneList);
            }
        }
        return result;
    }

    public void sortByFlightRange(List<Plane> planeList, Comparison comparison) {

        if (comparison == Comparison.INCREASE){
            planeList.sort(Comparator.comparing(Plane::getFlightRange));
        } else {
            planeList.sort(Comparator.comparing(Plane::getFlightRange).reversed());
        }
    }

    public int totalCapacity(List<Plane> planeList) {
        for (Plane aPlaneList : planeList) {
            totalCapacity += aPlaneList.getCapacity();
        }
        return totalCapacity;
    }

    public double totalLoadCapacity(List<Plane> planeList) {
        for (Plane aPlaneList : planeList) {
            totalLoadCapacity += aPlaneList.getLoadCapacity();
        }
        return totalLoadCapacity;
    }

    public List<Plane> getPlaneList() {
        return planeList;
    }

    public void setPlaneList(List<Plane> planeList) {
        this.planeList = planeList;
    }

    public int getTotalCapacity() {
        return totalCapacity;
    }

    public void setTotalCapacity(int totalCapacity) {
        this.totalCapacity = totalCapacity;
    }

    public double getTotalLoadCapacity() {
        return totalLoadCapacity;
    }

    public void setTotalLoadCapacity(double totalLoadCapacity) {
        this.totalLoadCapacity = totalLoadCapacity;
    }
}

File: PythonCourse/lab2/CompanyManager.py
from Plane import Plane


class CompanyManager:
    __total_capacity, __total_load_capacity = 0, 0

    def __init__(self, plane_list=None):
        self.plane_list = plane_list

    def search_fuel_consumption(self, fuel_consumption):
        result = []
        for plane in self.plane_list:
            if plane.fuel_consumption == int(fuel_consumption):
                result.append(plane)
        return result

    def sort_by_flight_range(self, comparison):
        if comparison.lower() == 'increase':
            self.plane_list = sorted(self.plane_list, key=lambda plane: plane.flight_range)
        else:
            self.plane_list = sorted(self.plane_list, key=lambda plane: plane.flight_range, reverse=True)

    def total_capacity(self):
        for plane in self.plane_list:
            self.__total_capacity += plane.capacity
        return self.__total_capacity

    def total_load_capacity(self):
        for plane in self.plane_list:
            self.__total_load_capacity += plane.load_capacity
        return self.__total_load_capacity
